lib/facility.rb

```diff
@@ -1,14 +1,74 @@
-class Facility
-  attr_reader :name, :address, :phone, :services
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+class Facility
+  
+  attr_reader :name, :address, :phone, :registered_vehicles,
+  :collected_fees
+  attr_accessor :services
+  def initialize(attributes = {})
+    @name = attributes[:name]
+    @address = attributes[:address]
+    @phone = attributes[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+  def register_vehicle(vehicle)
+    if @services.include?('Vehicle Registration')
+      vehicle.registration_date = Time.now
+      if vehicle.antique?
+        @collected_fees += 25
+        vehicle.plate_type = :antique
+      elsif vehicle.electric_vehicle?
+          @collected_fees += 200
+          vehicle.plate_type = :ev
+      else 
+        @collected_fees += 100
+        vehicle.plate_type = :regular
+        end
+      @registered_vehicles << vehicle
+    end
```

Comment:
For nested conditionals like this one, you could use an explicit return for a guard clause that would eliminate the need for nesting:
```ruby
return false unless @services.include?('Vehicle Registration')
vehicle.registration_date = Time.now
# etc
```